draco/tests/test_soft.py

```diff
@@ -332,8 +329,681 @@ def test_same_field():
     )
 
 
-def test_type_linear():
-    b = soft.blocks["type_linear"]
+def test_same_field_grt3():
+    b = soft.blocks["same_field_grt3"]
+    assert isinstance(b, Block)
+
+    assert (
+        list_preferences(
+            b.program
+            + """
+    entity(field,root,temperature).
+    entity(field,root,date).
+
+    entity(mark,v,m).
+    entity(encoding,m,e1).
+    entity(encoding,m,e2).
+
+    attribute((encoding,field),e1,temperature).
+    attribute((encoding,field),e2,date).
+    """
+        )
+        == []
+    )
+
+    assert (
+        list_preferences(
+            b.program
+            + """
+    entity(field,root,temperature).
+
+    entity(mark,v,m).
+    entity(encoding,m,e1).
+    entity(encoding,m,e2).
+
+    attribute((encoding,field),e1,temperature).
+    attribute((encoding,field),e2,temperature).
+    """
+        )
+        == []
+    )
+
+    assert (
+        list_preferences(
+            b.program
+            + """
+    entity(field,root,temperature).
+
+    entity(mark,v,m).
+    entity(encoding,m,e1).
+    entity(encoding,m,e2).
+    entity(encoding,m,e3).
+
+    attribute((encoding,field),e1,temperature).
+    attribute((encoding,field),e2,temperature).
+    attribute((encoding,field),e3,temperature).
+    """
+        )
+        == [("same_field_grt3", "temperature")]
+    )
+
+    assert (
+        list_preferences(
+            b.program
+            + """
+    entity(field,root,temperature).
+    entity(field,root,date).
+
+    entity(mark,v,m).
+
+    entity(encoding,m,e1).
+    entity(encoding,m,e2).
+    entity(encoding,m,e3).
+    entity(encoding,m,e4).
+    entity(encoding,m,e5).
+    entity(encoding,m,e6).
+
+    attribute((encoding,field),e1,temperature).
+    attribute((encoding,field),e2,temperature).
+    attribute((encoding,field),e3,temperature).
+    attribute((encoding,field),e4,date).
+    attribute((encoding,field),e5,date).
+    attribute((encoding,field),e6,date).
+    """
+        )
+        == [("same_field_grt3", "temperature"), ("same_field_grt3", "date")]
+    )
+
+    # use field temperature three times with mark m1,
+    # and field date with 2 different marks
+    assert (
```

Comment:
To simplify, we could remove temperature and only concentrate on the case with date and two marks. Simplifies the tests a bit.